state-chain/pallets/cf-elections/src/electoral_systems/change.rs

```diff
@@ -143,3 +143,70 @@ impl<
 		)
 	}
 }
+
+#[cfg(test)]
+mod test_change {
+
+	use crate::electoral_system::{mocks::MockElectoralSystem, ConsensusStatus};
+
+	thread_local! {
+		pub static HOOK_HAS_BEEN_CALLED: std::cell::Cell<bool> = const { std::cell::Cell::new(false) };
+	}
+
+	pub struct MockHook;
+	impl OnChangeHook<(), u64> for MockHook {
+		fn on_change(_id: (), _value: u64) {
+			HOOK_HAS_BEEN_CALLED.with(|hook_called| hook_called.set(true));
+		}
+	}
+
+	use super::*;
+	#[test]
+	fn consensus_not_possible_because_of_different_votes() {
+		let mut electoral_system =
+			MockElectoralSystem::<Change<(), u64, (), MockHook>>::new((), (), ());
+		let consensus = electoral_system
+			.new_election((), ((), 1), ())
+			.unwrap()
+			.check_consensus(None, vec![((), 1), ((), 5), ((), 3)], 3)
+			.unwrap();
+		assert_eq!(consensus, None);
+	}
+
+	#[test]
+	fn consensus_when_all_votes_the_same() {
+		let mut electoral_system =
+			MockElectoralSystem::<Change<(), u64, (), MockHook>>::new((), (), ());
+		let consensus = electoral_system
+			.new_election((), ((), 1), ())
+			.unwrap()
+			.check_consensus(None, vec![((), 1), ((), 1), ((), 1)], 3)
+			.unwrap();
+		assert_eq!(consensus, Some(1));
+	}
+
+	#[test]
+	fn not_enough_votes_for_consensus() {
+		let mut electoral_system =
+			MockElectoralSystem::<Change<(), u64, (), MockHook>>::new((), (), ());
+		let consensus = electoral_system
+			.new_election((), ((), 1), ())
+			.unwrap()
+			.check_consensus(None, vec![((), 1)], 3)
+			.unwrap();
+		assert_eq!(consensus, None);
+	}
+
+	#[test]
+	fn finalize_election() {
+		let mut electoral_system =
+			MockElectoralSystem::<Change<(), u64, (), MockHook>>::new((), (), ());
+		electoral_system
+			.new_election((), ((), 1), ())
+			.unwrap()
+			.set_consensus_status(ConsensusStatus::Changed { previous: 1, new: 2 });
+		electoral_system.finalize_elections(&()).unwrap();
+		// Hook should have been called
+		assert!(HOOK_HAS_BEEN_CALLED.with(|hook_called| hook_called.get()));
```

Comment:
This test can be more thorough:

- Create election with initial value
- Set consensus to the same as the initial value
- on_finalize => hook should not be called
- Set consensus to a different value
- on_finalize => hook should be called, election deleted.